container-search/src/main/java/com/yahoo/prelude/querytransform/QueryRewrite.java

```diff
@@ -200,6 +204,21 @@ private static void removeOtherNonrankedChildren(CompositeItem parent, int index
                 parent.removeItem(i);
         }
     }
+
+    private static boolean isRanked(Item item) {
+        if (item instanceof CompositeItem) {
+            boolean isRanked = false;
+            for (Item child : ((CompositeItem)item).items())
+                isRanked |= isRanked(child);
```

Comment:
if (isRanked(child)) return true;
No need to iterate all children if one is unhappy.